typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
As for how important it is, I don't know. Right now I don't think it matters at all, but for future developers using this as a framework I think it varies by implementation.

The reason I'd like to make it track with other changes to Room is that I think that is the natural assumption one would make who was coming to this code with no prior knowledge. I'd like to support that assumption insofar as I can, because I think it makes it more future-friendly. That's not necessarily a hard requirement, but just a value I'm factoring in when considering how to solve this problem. If it has to be sacrificed then that's a valid trade-off, but I think the gist above shows a way we can support it with relative ease.